### ISSUE TITLE
Compilation Error in oxc_transformer v0.48.2 on Rust

### ISSUE THREAD

**Issue by CoderSerio** [oxc-project/oxc#8913]
## Environment

- Rust version: stable
- Operating System: macOS
- oxc_transformer v0.48.2 (same issue occurs in v0.47.x)

## Error Info

When I run `cargo run`, it occurs: 

```bash
  Compiling oxc_transformer v0.48.2
error: `std::option::Option::<T>::unwrap` is not yet stable as a const fn
   --> /Users/carbon/.cargo/registry/src/index.crates.io-6f17d22bba15001f/oxc_transformer-0.48.2/src/jsx/refresh.rs:547:40
    |
547 | ... usize = base64_encoded_len(SHA1_HASH_LEN, true).unwrap();
    |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: could not compile `oxc_transformer` (lib) due to 1 previous error
warning: build failed, waiting for other jobs to finish...
error: failed to compile `arashi-builder v0.1.0 (/Users/carbon/Desktop/programms/good-open-source-project/arashi-builder)`, intermediate artifacts can be found at `/Users/carbon/Desktop/programms/good-open-source-project/arashi-builder/target`.
To reuse those artifacts with a future compilation, set the environment variable `CARGO_TARGET_DIR` to that path.
carbon@CarbondeMacBook-Air ~/Desktop/programms/good-open-source-project/arashi-builder (master+*) $ cargo install --path .
  Installing arashi-builder v0.1.0 (/Users/carbon/Desktop/programms/good-open-source-project/arashi-builder)
error: failed to compile `arashi-builder v0.1.0 (/Users/carbon/Desktop/programms/good-open-source-project/arashi-builder)`, intermediate artifacts can be found at `/Users/carbon/Desktop/programms/good-open-source-project/arashi-builder/target`.
To reuse those artifacts with a future compilation, set the environment variable `CARGO_TARGET_DIR` to that path.

Caused by:
  failed to run `rustc` to learn about target-specific information

Caused by:
  process didn't exit successfully: `/Users/carbon/.rustup/toolchains/stable-aarch64-apple-darwin/bin/rustc - --crate-name ___ --print=file-names -Zunstable-options -Zallow-features=const_eval_select --crate-type bin --crate-type rlib --crate-type dylib --crate-type cdylib --crate-type staticlib --crate-type proc-macro --print=sysroot --print=split-debuginfo --print=crate-name --print=cfg` (exit status: 1)
  --- stderr
  error: the option `Z` is only accepted on the nightly compiler

  help: consider switching to a nightly toolchain: `rustup default nightly`

  note: selecting a toolchain with `+toolchain` arguments require a rustup proxy; see <https://rust-lang.github.io/rustup/concepts/index.html>

  note: for more information about Rust's stability policy, see <https://doc.rust-lang.org/book/appendix-07-nightly-rust.html#unstable-features>

  error: 1 nightly option were parsed
```

## Reason

`std::option::Option::<T>::unwrap` is not yet stable as a const fn

## Remarks

I would like to try to fix it 💗

**Comment by overlookmotel** [comment]
What version of Rust are you using? The minimum version we support is listed in `Cargo.toml` - 1.80.

https://github.com/oxc-project/oxc/blob/0df939cb754198258346059f1661415b2ecb3e25/Cargo.toml#L14

We rely on clippy to let us know if we try to use features which aren't yet stable in that minimum Rust version. It's possible clippy is mistaken on this one, but I thought I'd check...

**Comment by Boshen** [comment]
Can you report `rustup show`?

**Comment by CoderSerio** [comment]
> Can you report `rustup show`?

I would like to do but I'm still on the way, so I can't provide it till a little later.

**Comment by CoderSerio** [comment]
> What version of Rust are you using? The minimum version we support is listed in `Cargo.toml` - 1.80.
> 
> [oxc/Cargo.toml](https://github.com/oxc-project/oxc/blob/0df939cb754198258346059f1661415b2ecb3e25/Cargo.toml#L14)
> 
> Line 14 in [0df939c](/oxc-project/oxc/commit/0df939cb754198258346059f1661415b2ecb3e25)
> 
>  rust-version = "1.80" # Support last 5 minor versions. 
> We rely on clippy to let us know if we try to use features which aren't yet stable in that minimum Rust version. It's possible clippy is mistaken on this one, but I thought I'd check...

Yes, and my Rust version is v1.80.1

<img width="360" alt="Image" src="https://github.com/user-attachments/assets/bf2d8941-d317-4e7f-a450-51ab67bec94d" />

**Comment by CoderSerio** [comment]
> Can you report `rustup show`?你能报告`rustup显示`？

<img width="402" alt="Image" src="https://github.com/user-attachments/assets/644c2613-3a7e-4800-b1a9-05621b5e7cff" />

**Comment by overlookmotel** [comment]
Sorry, I should have checked the docs. I trusted in clippy too much, and assumed most likely explanation was an old Rust version, since OP didn't specify what version they're using.

@CoderSerio Thanks for finding this.
